crypto/compactcert/builder.go

```diff
@@ -248,6 +261,7 @@ func (b *Builder) Build() (*Cert, error) {
 
 	c.SigProofs = *sigProofs
 	c.PartProofs = *partProofs
+	c.PositionsToReveal = revealsSequence
```

Comment:
Before returning, cache the cert, otherwise, `b.cert` is not used, and can be removed. 
```
b.cert = c
```